### ISSUE TITLE
tuned-ppd EPP setting on AC/DC

### ISSUE THREAD

**Issue by spandruvada** [redhat-performance/tuned#645]
This issue is for tuned-ppd.

The EPP is not set correctly as power-profiles-daemon was doing. On AC power-profiles-daemon was setting balanced-performance and on DC balanced-power (I think this is relatively new change in PPD).
But the new tuned-ppd package only sets balanced-performance always.

This commit from PPD, which implemented this:
The PPD commit 3b4249d8e90f24c3aa910af83ab349ebf23935e5 sets
balanced-power and  balanced-performance for DC and AC, respectively.

**Comment by zacikpa** [comment]
Hi! What version of TuneD are you running? This was implemented in the most recent version, 2.23.0, see #616.

**Comment by stereomato** [comment]
For me it does set up balance_power on battery (which is annoying due to https://gitlab.freedesktop.org/upower/power-profiles-daemon/-/issues/151 )

**Comment by yorickpeterse** [comment]
@stereomato You can fix that using the following `/etc/tuned/ppd.conf`:

```
[main]
# The default PPD profile
default=balanced
battery_detection=true

[profiles]
# PPD = TuneD
power-saver=powersave
balanced=balanced
performance=throughput-performance

[battery]
# PPD = TuneD
balanced=balanced
```

You then need to restart tuned and (I _think_) tuned-ppd.

This ensures that when you're on battery power, the energy performance preference is set to `balance_performance` instead of `balance_power`.

**Comment by zacikpa** [comment]
@yorickpeterse yes, exactly, thank you for the response. Just to clarify, only `tuned-ppd` needs to be restarted for the change to take effect.

Alternatively, one can simply leave out the `[battery]` section from the configuration file and set `battery_detection=false`; the result will be the same.

@stereomato Using the configuration file, you can also experiment with other mappings to TuneD profiles, if the default ones do not suit you.
